### PR TITLE
Update msftidy to error when module super class is incorrect

### LINKED ISSUE #6365: Update msftidy to enforce that modules of specific type (auxiliary, post, exploit, etc) use the correct class
Today it seems like this is not enforced and modules in auxiliary can use Msf::Exploit::Remote, which is a little confusing and may be problematic.

### PR DESCRIPTION
**Pull request by jhart-r7**
Fixes #6365.

Currently all modules pass this check, but when I modified one to be invalid, I get this:

```
modules/auxiliary/dos/http/apache_commons_fileupload_dos.rb - [ERROR] Invalid super class for auxiliary module (found 'Msf::Exploit::Remote', expected something like (?-mix:Msf::Auxiliary$)
```

This also refactors msftidy to be testable and adds some initial tests for it, which seems necessary given what we use it for.

### CODE REVIEW

File: tools/dev/msftidy.rb [hunk 1]
```diff
@@ -65,6 +65,7 @@ class Msftidy
 
   def initialize(source_file)
     @full_filepath = source_file
+    @module_type = File.dirname(File.expand_path(@full_filepath))[/\/modules\/([^\/]+)/, 1]
```
Comment (jhart-r7) [inline tools/dev/msftidy.rb]: I'm not thrilled with this, but I'm not sure of a way to determine the _intended_ module type other than by looking at the directory it lives in.  This assumes any module passed to `msftidy` is under a `modules` directory.

**Comment by jhart-r7** [comment]
The failure reported in https://github.com/jhart-r7/metasploit-framework/commit/5ac4e9aa6bd0bf2efc4b9f5cac31a72ea53e2a46 seems to be unrelated.  Re-running.

**Comment by jhart-r7** [comment]
Marking this delayed.  The unit tests work but adding to them is painful.  Reworking.

**Comment by jhart-r7** [comment]
I reworked the tests.  This should be mostly landable now.

**Comment by bcook-r7** [comment]
Updated for master, thanks @jhart-r7

**Comment by bcook-r7** [comment]
I'm making a small change so this is compatible with #6405:
cc436fe4380fa12862e9eb6dd60f58bfb3086b1a

**Pull request merged**